### PR TITLE
[argocd-notifications] add monitoring and slack bot service annotations

### PR DESCRIPTION
**Pull request by maciekm**
Checklist:

* [x] I have update the chart version in `Chart.yaml` following Semantic Versioning.
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have followed the testing instructions in the [contributing guide](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md).
* [x] I have signed the CLA and the build is green.
* [x] I will test my changes again once merged to master and published.

This PR
* Adds monitoring to the controller
* Adds `logLevel` option
* Adds slack bot service annotations allowing external DNS registration, e.g.:
  ```
  annotations:
    external-dns.alpha.kubernetes.io/hostname:  argocd-notifications-bot.example.com
  ```

documentation:
* argocd-notifications metrics -> https://github.com/argoproj-labs/argocd-notifications/blob/master/docs/monitoring.md

### CODE REVIEW

File: charts/argocd-notifications/templates/deployment.yaml [hunk 1]
```diff
@@ -29,6 +29,14 @@ spec:
...
             - /app/argocd-notifications
             - controller
+            - --loglevel={{ .Values.logLevel }}
+            {{- if .Values.metrics.enabled }}
+            - --metrics-port={{ .Values.metrics.port }}
+          ports:
+          - containerPort: {{ .Values.metrics.port }}
+            name: metrics
+            protocol: TCP
+            {{- end }}
```
Comment (andyfeller) [inline charts/argocd-notifications/templates/deployment.yaml:39]: I think I understand what is going on here though it took me a double take.  How would you feel about separating out the conditional between the command and ports?
```suggestion
            {{- if .Values.metrics.enabled }}
            - --metrics-port={{ .Values.metrics.port }}
            {{- end }}
          ports:
          {{- if .Values.metrics.enabled }}
          - containerPort: {{ .Values.metrics.port }}
            name: metrics
            protocol: TCP
          {{- end }}
```

File: charts/argocd-notifications/values.yaml [hunk 2]
```diff
@@ -87,6 +87,18 @@ secret:
       # email address in from field
       from:
 
+logLevel: info
+
+metrics:
+  enabled: false
+  port: 9001
```
Comment (andyfeller) [inline charts/argocd-notifications/values.yaml:94]: I'm not familiar with the metrics support you're adding here.  Is there anything within the project documentation to link back for those interested in exploring more?

Comment (maciekm) [reply charts/argocd-notifications/values.yaml:94]: Thanks @andyfeller for the review. Updated description with a link to the metrics doc

File: charts/argocd-notifications/templates/deployment.yaml (hunk 1 above)
Comment (maciekm) [reply charts/argocd-notifications/templates/deployment.yaml]: updated, thanks

Comment (andyfeller) [reply charts/argocd-notifications/templates/deployment.yaml]: apologies but not seeing the updates.  as soon as I do, I will get them reviewed and shipped

File: charts/argocd-notifications/templates/deployment.yaml [hunk 3]
```diff
@@ -29,6 +29,16 @@ spec:
...
+            - --loglevel={{ .Values.logLevel }}
+            {{- if .Values.metrics.enabled }}
+            - --metrics-port={{ .Values.metrics.port }}
+            {{- end }}
+          ports:
+          {{- if .Values.metrics.enabled }}
+          - containerPort: {{ .Values.metrics.port }}
+            name: metrics
+            protocol: TCP
+          {{- end }}
```
Comment (maciekm) [inline charts/argocd-notifications/templates/deployment.yaml:41]: @andyfeller, I did `force-push`, sorry for that. Here is that change

File: charts/argocd-notifications/templates/deployment.yaml (hunk 1 above)
Comment (maciekm) [reply charts/argocd-notifications/templates/deployment.yaml]: @andyfeller this thread is `Outdated`, do you see the changes in `FilesChanged`?

**Pull request merged**